app/controllers/devise_recognisable/sessions_controller.rb

```diff
@@ -3,23 +3,27 @@
 class DeviseRecognisable::SessionsController < Devise::SessionsController
   prepend_before_action :check_for_authentication_token, only: :new
   prepend_before_action :perform_ip_check, only: :create
+  append_after_action :store_recognisable_details, only: :create
  
   def perform_ip_check
     # Find the user
     self.resource = resource_class.find_by(email: params[resource_name][:email])
-    return unless self.resource && self.resource.last_sign_in_ip.present?
+    previous_session = Devise.ref('DeviseRecognisable::RecognisableSession').get
+      .where( recognisable: self.resource ).last
```

Comment:
You need to order the results otherwise they might not be returned in date order. Suggest ordering by `created_at: :desc` and then getting the `.first`